v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -1171,6 +1172,105 @@ func assignMapFromMap(
 	}, nil
 }
 
+// assignUserAssignedIdentityMapFromArray will generate a code fragment to populate a userAssignedIdentity array from
+// a map whose key is the ARM ID of the userAssignedIdentity
+// <arr> := make([]<arrType>, 0, len(source.UserAssignedIdentities))
+//
+//	for key, _ := range source.UserAssignedIdentities {
+//	   ref := genruntime.CreateResourceReferenceFromARMID(key)
+//	   <arr> = append(<arr>, UserAssignedIdentityDetails{Reference: ref})
+//	}
+//
+// <writer> = <arr>
+func assignUserAssignedIdentityMapFromArray(
+	sourceEndpoint *TypedConversionEndpoint,
+	destinationEndpoint *TypedConversionEndpoint,
+	conversionContext *PropertyConversionContext) (PropertyConversion, error) {
+
+	// Require both source and destination to not be bag items
+	if sourceEndpoint.IsBagItem() || destinationEndpoint.IsBagItem() {
+		return nil, nil
+	}
+
+	// Require source to be a map type
+	_, sourceIsMap := astmodel.AsMapType(sourceEndpoint.Type())
+	if !sourceIsMap {
+		return nil, nil
+	}
+
+	// Require destination to be an array type
```

Comment:
Should you check the array time has the expected element type?

Reply:
Have added a check that the destination is a `TypeName`